assertk/src/jvmMain/kotlin/assertk/assertions/any.kt

```diff
@@ -82,7 +82,7 @@ fun <T : Any> Assert<T>.isDataClassEqualTo(expected: T) = given { actual ->
 private fun <T> Assert<T>.isDataClassEqualToImpl(expected: T, kclass: KClass<*>?): Unit = given { actual ->
     if (actual == expected) return
     val compareProps = actual != null && expected != null
-    if (compareProps && kclass != null && kclass.isData) {
+    if (compareProps && kclass != null && kclass.isData && kclass.objectInstance == null) {
```

Comment:
This seemed like the best way to check if `kclass` is a `data object`, but there could be a better way.

Reply:
could we check the types directly? This should always fail if `expected::class != actual::class` right?

Reply:
Sure, how does this look?
```kotlin
    val compareProps = actual != null && expected?.let { it::class } == kclass
    if (compareProps && kclass != null && kclass.isData) {
```